### PR TITLE
[READY] Ash Drake bug fixes and edits

### LINKED ISSUE #41887: Ash Drake Arena can select inaccessible tiles as the safe tile
For example, if fighting the Ash Drake in the Heirophant Arena, the safe zone for the arena attack can be on one of the indestructible pillars.

### PR DESCRIPTION
**Pull request by Whoneedspacee**
:cl: Joe Berry
add: Mass fire attack, sends fire out from the ash drake in all directions
add: Adds an enraged attack for ash drake, heals him as well as making him glow and go faster, spawning massive amounts of fire in all directions
del: Removes the old triple swoop with lava pools attack
tweak: Lava pools can now spawn with the normal fire breath attack sometimes
tweak: Lava pools now have changed delays for lesser amounts so they don't all just place around one area
tweak: Increases default swoop delay for #41178 
balance: Teleporting out of the lava arena now has some actual consequences by enraging the ash drake
fix: Makes lava arena a bit less laggy by not recalculating range_turfs every time
fix: Fixes #41887 though this will not change the turfs to basalt temporarily to prevent moving through indestructible walls
fix: Fire lines would not spawn if their range would place their final turf location outside of the map
fix: The arena attack will no longer destroy indestructible open turfs
/:cl:

[why]: Just some bug fixes and prevents some cheese weirdness when people teleport out of the lava arena. The lava arena attack would continue going even when there was nobody in the arena. Also increasing the delays on the swoop attacks makes it feel less like it's chaining attacks like lava arena over and over. The triple swoop with lava pools was tedious as best and I think this attack is more massively threatening for a second phase attack.

[Current Version Video](https://www.youtube.com/watch?v=qkQk0-NhXOw&feature=youtu.be)

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm [hunk 1]
```diff
@@ -144,63 +146,97 @@ Difficulty: Medium
...
 				new /obj/effect/temp_visual/lava_safe(T)
 		amount--
 		sleep(24)
-
-/mob/living/simple_animal/hostile/megafauna/dragon/proc/fire_cone()
+	return 1 // attack finished completely
+
+/mob/living/simple_animal/hostile/megafauna/dragon/proc/arena_escape_enrage() // you ran somehow / teleported away from my arena attack now i'm mad fucker
+	SetRecoveryTime(80)
+	visible_message("<span class='boldwarning'>[src]'s starts to glow vibrantly as its wounds seem to heal!</span>")
```
[the remark this answers is not in the document]
Comment (Whoneedspacee) [reply code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm]: oops forgot to change it cause i was shiftin around messages

**Comment by KomradeSpectre** [comment]
Idea for a PR that complements this, megafauna should target or be enraged at people using chameleon projectors nearby them.

**Comment by Whoneedspacee** [comment]
Shouldn’t be too hard to do, I’ll put it on my mental list. I’ve been meaning to edit the base megafauna class and do some general QoL for megafauna. Maybe give them all unique devours as well that typa stuff.

**Comment by KomradeSpectre** [comment]
Though I really shouldn't be discussing it here, BDM needs some way of getting over lava, if not on a cooldown. Currently you can lure BDM to a specific spot at southern lavaland, and outrange him with APCs with no attack he can do back that hits you. You can literally sit in one spot.

**Comment by Whoneedspacee** [comment]
Really just message me on discord if you wanna, and yeah I was gonna give BDM the tendril wings.

**Pull request merged**